### PR TITLE
feat: add row echelon matrix algorithm

### PR DESCRIPTION
**Pull request by piyushk77**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Maths/RowEchelon.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  }
+}
+
+const formatResult = (matrix) => {
+  let precision = 5
+  let numRows = matrix.length
+  let numCols = matrix[0].length
+  for (let i = 0; i < numRows; i++) {
+    for (let j = 0; j < numCols; j++) {
+      matrix[i][j] = parseFloat(matrix[i][j].toFixed(precision))
```
Comment (appgurueu) [inline Maths/RowEchelon.js:97]: Not a fan of this rounding. Note that you should be able to specify a float equality tolerance in the tests.

**Review by appgurueu — COMMENTED** [review]
The decomposition into a set of properly named small functions makes this reasonably readable even without comments everywhere, though I suppose some comments wouldn't hurt. The doc comment needs to be moved to the proper function though, and the error case should `throw`. `formatResult` should be removed in favor of just using float tolerance in testing.

File: Maths/RowEchelon.js (hunk 1 above)
Comment (piyushk77) [reply Maths/RowEchelon.js:97]: Got it. Changing it to use float tolerance instead of rounding.

Comment (piyushk77) [reply Maths/RowEchelon.js:97]: Got it. Changing it to use float tolerance instead of rounding.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**